pkg/nettools/nettools.go

```diff
@@ -318,53 +301,27 @@ func SetupContainerSideNetwork(info *types.Result) (*ContainerNetwork, error) {
 		return nil, fmt.Errorf("failed to set %q up: %v", tapInterfaceName, err)
 	}
 
-	dhcpVeth := &netlink.Veth{
-		LinkAttrs: netlink.LinkAttrs{
-			Name:  dhcpVethContainerEndName,
-			Flags: net.FlagUp,
-			MTU:   contVeth.Attrs().MTU,
-		},
-		PeerName: dhcpVethDhcpEndName,
-	}
-	if err := netlink.LinkAdd(dhcpVeth); err != nil {
-		return nil, fmt.Errorf("failed to create dhcp veth pair: %v", err)
-	}
-	if err = netlink.LinkSetUp(dhcpVeth); err != nil {
-		return nil, fmt.Errorf("failed to set %q up: %v", dhcpVethContainerEndName, err)
-	}
-
-	dhcpSideVeth, err := netlink.LinkByName(dhcpVethDhcpEndName)
+	br, err := SetupBridge(containerBridgeName, []netlink.Link{contVeth, tap})
 	if err != nil {
-		return nil, fmt.Errorf("failed to find dhcp side veth before moving it: %v", err)
+		return nil, fmt.Errorf("failed to create bridge: %v", err)
 	}
 
-	if err := netlink.LinkSetNsFd(dhcpSideVeth, int(dhcpNS.Fd())); err != nil {
-		return nil, fmt.Errorf("failed to move dhcp side veth to its ns: %v", err)
+	if err = netlink.AddrAdd(br, mustParseAddr(internalDhcpAddr)); err != nil {
+		return nil, fmt.Errorf("failed to set address for the bridge: %v", err)
 	}
 
-	if err := dhcpNS.Do(func(ns.NetNS) error {
-		veth, err := netlink.LinkByName(dhcpVethDhcpEndName)
-		if err != nil {
-			return fmt.Errorf("failed to find dhcp side veth: %v", err)
-		}
-		if err = netlink.LinkSetUp(veth); err != nil {
-			return fmt.Errorf("failed to set %q up: %v", dhcpVethDhcpEndName, err)
-		}
-		if err = netlink.AddrAdd(veth, mustParseAddr(internalDhcpAddr)); err != nil {
-			return fmt.Errorf("failed to set address for dhcp side veth: %v", err)
+	// block DHCP traffic from/to CNI-provided link
+	for _, item := range []struct{ chain, opt string }{
```

Comment:
Additional spaces around struct elements.
Linter (probably `go vet`) will help to catch such things.